### DISCUSSION TITLE
No finality of transcribing for replaced phrases in STT Streaming

### DISCUSSION THREAD

**Discussion by espdev** [deepgram/community#658] [General help]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

We use the STT streaming API (nova-2 model). We want to define [replaced phrases](https://developers.deepgram.com/docs/find-and-replace), for example:
```
replace=hello:hello my friend
replace=stuff:stop
```

When we say a phrase that should be replaced, Deepgram makes the replacement, but does not send a finality flag until we say something different from the replaced phrase.

For example, we say "hello" (out log from the event handler):
```
TRANSCRIPT: '', is_final=False
TRANSCRIPT: '', is_final=False
TRANSCRIPT: 'hello my friend', is_final=False    # <- we say "hello"
TRANSCRIPT: 'hello my friend', is_final=False
TRANSCRIPT: 'hello my friend', is_final=False
TRANSCRIPT: 'hello my friend', is_final=False
TRANSCRIPT: 'hello my friend', is_final=False
TRANSCRIPT: 'hello my friend', is_final=False
TRANSCRIPT: 'hello my friend', is_final=False
TRANSCRIPT: 'hello my friend', is_final=False
TRANSCRIPT: 'hello my friend goodbye', is_final=False  # <- we say "goodbay"
TRANSCRIPT: 'hello my friend goodbye', is_final=True
TRANSCRIPT: '', is_final=False
FINAL RESULT: hello my friend goodbye
```

Deepgam will endlessly send us the replaced phrase "hello my friend" with `result.is_final=False` until we say something else, then the transcript will be finalized.

It is impossible to use the "replace" function because of the behavior.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.
